include/drivers/ipm.h

```diff
@@ -221,6 +346,185 @@ static inline int z_impl_ipm_set_enabled(const struct device *ipmdev,
 	return api->set_enabled(ipmdev, enable);
 }
 
+#ifdef CONFIG_IPM_MULTICHANNEL_DT_SUPPORT
+
+/**
+ * @brief Try to send a message over the IPM device through a specific channel.
+ *
+ * If the hardware supports it, the message is sent through a specific channel.
+ *
+ * @param ipmdev Driver instance
+ * @param channel IPM device channel
+ * @param wait If nonzero, busy-wait for remote to consume the message. The
+ *	       message is considered consumed once the remote interrupt handler
+ *	       finishes. If there is deferred processing on the remote side,
+ *	       or you would like to queue outgoing messages and wait on an
+ *	       event/semaphore, you can implement that in a high-level driver
+ * @param id Message identifier. Values are constrained by
+ *        @a ipm_max_data_size_get since many boards only allow for a
+ *        subset of bits in a 32-bit register to store the ID.
```

Comment:
What's actually the difference between channel and id in this API? I believe both parameters are intended to use some hardware mechanism to distinguish separated "streams" of data through single hardware peripheral. Is there a need to keep both in the same function?

Reply:
I tried to explain my interpretation in the comment here https://github.com/zephyrproject-rtos/zephyr/pull/38523#issuecomment-919127028